### PR TITLE
Add documentation for reverse()

### PR DESCRIPTION
**Pull request by mcol**
#### Submission Checklist

- [X] Builds locally 
- [X] Declare copyright holder and open-source license: see below

#### Summary
This adds docs for `reverse`, which was added in https://github.com/stan-dev/math/pull/1650. I'm not entirely convinced of the section titles I used, I followed what was around and couldn't think of something better.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Marco Colombo

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/functions-reference/array_operations.Rmd [hunk 1]
```diff
@@ -409,3 +410,21 @@ Number of components of v less than v[s]
 `int` **`rank`**`(int[] v, int s)`<br>\newline
 Number of components of v less than v[s]
 
+## Reversing Functions {#reversing-functions}
+
+Reversing allows to create a copy of array in which elements are in reverse
```
Comment (bob-carpenter) [inline src/functions-reference/array_operations.Rmd]: This isn't quite grammatical and it's not a copy.  Also, we want to make the doc more active.  So this should be:

```
Stan provides functions to create a new array by reversing the order of elements in an existing array.  For example, ...
```

File: src/functions-reference/array_operations.Rmd [hunk 2]
```diff
@@ -409,3 +410,21 @@ Number of components of v less than v[s]
...
+\index{{\tt \bfseries reverse }!{\tt (real[] v): real[]}|hyperpage}
+
+`real[]` **`reverse`**`(real[] v)`<br>\newline
+Create a copy of the array with the elements in reverse order.
+
+<!-- int[]; reverse; (int[] v); -->
+\index{{\tt \bfseries reverse }!{\tt (int[] v): int[]}|hyperpage}
+
+`int[]` **`reverse`**`(int[] v)`<br>\newline
+Create a copy of the array with the elements in reverse order.
```
Comment (bob-carpenter) [inline src/functions-reference/array_operations.Rmd]: Does this only work for integer and real arrays?  It should work for any array if possible, or if not, at least all arrays up to a given number of dimensions.  If this PR doesn't create a generic `reverse` function, please create a new issue with that goal.  Then the doc can just use `T[] reverse (T[] x)`.

File: src/functions-reference/matrix_operations.Rmd [hunk 3]
```diff
@@ -1611,3 +1612,16 @@ Number of components of v less than v[s]
 `int` **`rank`**`(row_vector v, int s)`<br>\newline
 Number of components of v less than v[s]
 
+## Reverse Functions {#reverse-functions}
+
+<!-- vector; reverse; (vector v); -->
+\index{{\tt \bfseries reverse }!{\tt (vector v): vector}|hyperpage}
+
+`vector` **`reverse`**`(vector v)`<br>\newline
+Create a copy of the vector with the elements in reverse order.
```
Comment (bob-carpenter) [inline src/functions-reference/matrix_operations.Rmd]: This isn't a copy any more than sorting is a copy.  It returns a new vector containing the elements of the argument vector in reverse order.

File: src/functions-reference/array_operations.Rmd [hunk 4]
```diff
@@ -409,3 +410,21 @@ Number of components of v less than v[s]
...
+Reversing allows to create a copy of array in which elements are in reverse
+order. For example, if `v` is declared as a real array of size 3, with values
+\[ \text{v} = (1,\, -10.3,\, 20.987), \] then
+\[ \mathrm{reverse(v)} = (20.987,\, -10.3,\, 1). \]
+
+<!-- real[]; reverse; (real[] v); -->
+\index{{\tt \bfseries reverse }!{\tt (real[] v): real[]}|hyperpage}
+
+`real[]` **`reverse`**`(real[] v)`<br>\newline
+Create a copy of the array with the elements in reverse order.
```
Comment (bob-carpenter) [inline src/functions-reference/array_operations.Rmd]: A copy would have the elements in the same order.  This function returns a new array with the argument array's elements in reverse order.

File: src/functions-reference/array_operations.Rmd (hunk 2 above)
Comment (mcol) [reply src/functions-reference/array_operations.Rmd]: The implementation does not restrict what can be reversed, so I'll change that to use `T[]`.

File: src/functions-reference/array_operations.Rmd (hunk 1 above)
Comment (mcol) [reply src/functions-reference/array_operations.Rmd]: Thanks for the much improved wording!

File: src/functions-reference/array_operations.Rmd [hunk 5]
```diff
@@ -334,7 +335,7 @@ Any mismatches will cause an error to be thrown.
  x3 = append_array(x1, x2);
 ```
 
-## Sorting functions {#sorting-functions}
+## Sorting Functions {#sorting-functions}
```
Comment (bob-carpenter) [inline src/functions-reference/array_operations.Rmd]: This should be `Sorting functions` not `Sorting Functions`.  

* `# XXX` uses title punctuation for `XXX` (all caps other than "function" words [prepositions, determiners, auxliary verbs, other particles, etc.])
* `## XXX` should use sentence punctuation for `XXX` (initial caps, not all caps)

File: src/functions-reference/array_operations.Rmd [hunk 6]
```diff
@@ -409,3 +410,16 @@ Number of components of v less than v[s]
 `int` **`rank`**`(int[] v, int s)`<br>\newline
 Number of components of v less than v[s]
 
+## Reversing Functions {#reversing-functions}
```
Comment (bob-carpenter) [inline src/functions-reference/array_operations.Rmd]: fix case

**Pull request merged**